service/tasks/machine.py

```diff
@@ -179,6 +149,42 @@ def start_machine_imaging(machine_request, delay=False):
     return async
 
 
+def enable_image_validation(machine_request, init_task, final_task, original_status="", error_handler_task=None):
+    if not error_handler_task:
+        error_handler_task = machine_request_error.s(machine_request.id)
+    # Task 3 = Validate the new image by launching an instance
+    admin_ident = machine_request.new_admin_identity()
+    admin_driver = get_admin_driver(machine_request.new_machine_provider)
+    if 'validating' in original_status:
+        image_id = machine_request.new_machine.identifier
+        celery_logger.info("Start with validating:%s" % image_id)
+        # If validating, seed the image_id and start here..
+        validate_task = validate_new_image.s(image_id, machine_request.id)
+        init_task = validate_task
+    else:
+        validate_task = validate_new_image.s(machine_request.id)
+        init_task.link(validate_task)
+    #Validate task returns an instance_id
+    # Task 4 = Wait for new instance to be 'active'
+    wait_for_task = wait_for_instance.s(
+        # NOTE: 1st arg, instance_id, passed from last task.
```

Comment:
FYI: `instance_id` is not an argument at all, and the `instance_id` variable defined on line 100 isn't actually used anywhere again. Are the comments out of date or is something missing?